Apsitvarkom.DataAccess/IRepository.cs

```diff
@@ -15,4 +15,9 @@ public interface IRepository<T> where T: class
     /// <param name="propertyCondition">Condition for the requested record acquisition.</param>
     /// <returns>Null if the instance was not found. Otherwise the acquired <see cref="T" /> instance.</returns>
     public Task<T?> GetByPropertyAsync(Expression<Func<T, bool>> propertyCondition);
+
+    /// <summary>Inserts a single <see cref="T" /> record to the data source.</summary>
+    /// <param name="modelToInsert">Model to be inserted into the repository.</param>
+    /// <returns>The inserted <see cref="T" /> model.</returns>
+    public Task<T> InsertAsync(T modelToInsert);
```

Comment:
Why do we need this method to return the same object we pass to it?

Reply:
Some food for thought about designing RESTful APIs (what we're doing here):
https://stackoverflow.com/a/28951049/13135665
https://softwareengineering.stackexchange.com/a/314069

Reply:
Maybe you're right, I initially thought that maybe the `InsertAsync` could mutate the model somehow, but on the other hand I don't think it should be ever do that, the point of this method is to insert the model to the database.

Reply:
The first comment still stands as why we're returning the created model from the endpoint, although we shouldn't change the model in this repository method (I will make necessary changes).

Reply:
See this commit https://github.com/vu-vibedosa/apsitvarkom-be/pull/101/commits/021b1009b57538ec13234860f2be60f4b231affe

Reply:
Yup. Meant exactly that. We shouldn't return it from the repository insert method, but as a CRUD Create action - sure. This would also allow us to add it to the polluted locations list on the front-end without having to get all locations again I believe.

Reply:
The first link had a nice reference though :D Saved it to my bookmarks.